core/src/main/scala/weaponregex/parser/Parser.scala

```diff
@@ -187,18 +218,30 @@ abstract class Parser(val pattern: String) {
     * @example
     *   `"\ u0020"`
     */
-  def charUnicode[A: P]: P[MetaChar] = Indexed("\\u" ~ CharIn("0-9a-zA-Z").rep(exactly = 4).!)
+  def charUnicode[A: P]: P[MetaChar] = Indexed("\\u" ~ hexDigit.rep(exactly = 4).!)
     .map { case (loc, hexDigits) => MetaChar("u" + hexDigits, loc) }
 
-  /** Parse a character with hexadecimal value with braces `\x{h...h}` (Character.MIN_CODE_POINT <= 0xh...h <=
-    * Character.MAX_CODE_POINT)
+  /** Parse a character with a code point `\x{h...h}`, where Character.MIN_CODE_POINT <= 0xh...h <=
+    * Character.MAX_CODE_POINT and x is [[weaponregex.parser.Parser.codePointEscChar]]
     * @return
     *   [[weaponregex.model.regextree.MetaChar]] tree node
     * @example
-    *   `"\x{0123}"`
+    *   `"\ x{0123}"` or `"\ u{0123}"`
+    * @see
+    *   [[weaponregex.parser.Parser.codePointEscChar]]
     */
-  def charHexBrace[A: P]: P[MetaChar] = Indexed("""\x{""" ~ CharIn("0-9a-zA-Z").rep(1).! ~ "}")
-    .map { case (loc, hexDigits) => MetaChar("x{" + hexDigits + "}", loc) }
+  def charCodePoint[A: P]: P[MetaChar] = Indexed(s"\\$codePointEscChar" ~ ("{" ~ hexDigit.rep(1) ~ "}").!)
```

Comment:
Perhaps a bit out of scope but given that the changes of #167 got pulled in here as well:

I think there's one more small problem with this part of the parser. In particular, from my testing on [regex101.com](https://regex101.com/) with JVM (using `"\x{h..h}"`) and JS (using `/\u{h..h}/u`) I found that it only accepts up-to a 5-digit (hexadecimal) number. So, I think this should be:

```suggestion
  def charCodePoint[A: P]: P[MetaChar] = Indexed(s"\\$codePointEscChar" ~ ("{" ~ hexDigit.rep(min=1, max=5) ~ "}").!)
```

However, both JVM and JS seem to be fine with any number of leading zeros, so technically it should be something like:

```suggestion
  def charCodePoint[A: P]: P[MetaChar] = Indexed(s"\\$codePointEscChar" ~ ("{" ~ "0".rep(min=0) ~ hexDigit.rep(min=1, max=5) ~ "}").!)
```

Reply:
Thanks for the bug report 👍 but technically it should accept upto 6 hex digits (without leading zeros) since a code point can be anything `0 <= codepoint <= 0x10FFFF` ([JS reference](https://tc39.es/ecma262/multipage/ecmascript-language-lexical-grammar.html#prod-CodePoint), [Java reference](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Pattern.html#characters))